org.scala-ide.sdt.core/src/org/scalaide/core/internal/builder/zinc/SbtInputs.scala

```diff
@@ -53,12 +54,21 @@ class SbtInputs(installation: IScalaInstallation,
   def analysisMap(f: File): Maybe[Analysis] =
     if (f.isFile)
       Maybe.just(Analysis.Empty)
-    else
-      allProjects.find(_.sourceOutputFolders.map(_._2.getLocation().toFile()).toSeq contains f) map (_.buildManager) match {
-        case Some(sbtManager: EclipseSbtBuildManager) => Maybe.just(sbtManager.latestAnalysis(incOptions))
-        case _                                     => Maybe.just(Analysis.Empty)
+    else {
+      def suchProjectWhereFileIsTheProduct(project: IScalaProject): Boolean = {
```

Comment:
I don't like the name. Since it returns Boolean name should follow convention for such methods.
Also it's not so easy to read for me.
Maybe something like 'containsProjects'? 
But I think it's a bit over-complicated overall. You can replace that whole `else` block with just:
`project.sourceOutputFolders exists(_._2.getLocation.toFile == f)`
(You can still use pattern match to avoid _2 of course if you like it more).
This way you also don't need to map all folders. If you want to make it more readable you can always assign result to val and return it.